pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java

```diff
@@ -404,4 +427,35 @@ public void close() {
         private static final Logger log = LoggerFactory.getLogger(ConsumerSocket.class);
 
     }
+
+    private static class DataConsistencyChecker {
+        private final ConcurrentHashMap<String, AtomicLong> messagesPerKey = new ConcurrentHashMap<>();
+        private final boolean failOnDuplicateKey;
+
+        public DataConsistencyChecker(boolean failOnDuplicateKey) {
+            this.failOnDuplicateKey = failOnDuplicateKey;
+        }
+
+        public void messageReceived(String key) throws Exception {
+            AtomicLong count = messagesPerKey.computeIfAbsent(key, k -> new AtomicLong());
+            long newValue = count.incrementAndGet();
+            if (failOnDuplicateKey && newValue == 2) {
+                String message = "Key '"+key+"' has been received more than once";
+                log.error(message);
+                throw new Exception(message);
```

Comment:
As long as there are duplicate messages, an exception will be thrown, there is no way to count, there will always be only one result.
Can some unit tests be added?

Reply:
we cannot do it right now
see
https://github.com/apache/pulsar/issues/9865